turbopy/core.py

```diff
@@ -402,7 +402,8 @@ def exchange_resources(self):
         This is the function where you call publish_resource, to tell
         other physics modules about data you want to share.
         """
-        pass
+        self.publish_resource({attribute: self.__dict__[attribute] for attribute in self.__dict__
```

Comment:
Currently this is being shared with the key set to be the name of the attribute. What happens if two different modules have attributes with the same name? I think there needs to be a way to distinguish them. Maybe use the class name (`self.__class__`) together with the attribute name?